src/item_repository.py

```diff
@@ -0,0 +1,102 @@
+
+import sqlite3
+import csv
+
+class ItemRepository:
+
+  def __init__(self) -> None:   
+    self.db_path = './todo.db'
+    self.connection = None
+
+
+  def connect_db(self):
+    if self.connection is None:
+      self.connection =  sqlite3.connect(self.db_path, check_same_thread=False)
+
+
+  def get_all_items(self):
+    try:
+      self.connect_db()
+      cursor = self.connection.cursor()
+      rows = cursor.execute('select * from items')
+      return rows
+    except Exception as e:
+      raise Exception('Error: ', e)
+
+  def get_item(self,item):
+    try:
+      self.connect_db()
+      cursor = self.connection.cursor()
+      rows = cursor.execute('select * from items where id=?',(item,))
+      return rows
+    except Exception as e:
+      raise Exception('Error: ', e)
+
+  def delete_item(self, item):
+    try:
+      self.connect_db()
+      cursor = self.connection.cursor()
+      cursor.execute('select count(id) from items where id = ?',(item,))
+      val = cursor.fetchone()[0]
+      if  val ==0 :
+        return {}
+      delete_cursor = cursor.execute('delete from items where id = ?', (item,))
+      self.connection.commit()
+      return "Delete was successful"
```

Comment:
the function can be refactored.
It is performing two sql statements, but the function name is just `delete_item`